### PR TITLE
Handle UnsupportedCharsetException in HttpLoggingInterceptor

### PR DESCRIPTION
**Pull request by dave-r12**
This is an attempt to push #1977 over the finish line.

### CODE REVIEW

File: okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java [hunk 1]
```diff
@@ -227,7 +228,15 @@ public Level getLevel() {
         Charset charset = UTF8;
         MediaType contentType = responseBody.contentType();
         if (contentType != null) {
-          charset = contentType.charset(UTF8);
+          try {
+            charset = contentType.charset(UTF8);
+          } catch (UnsupportedCharsetException e) {
+            logger.log("");
+            logger.log("Couldn't decode the response body; charset is likely malformed.");
```
Comment (swankjesse) [inline okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java]: maybe include the MediaType so the unlucky user can see `UFT-8` or whatever ?

**Comment by swankjesse** [comment]
LGTM!

File: okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java (hunk 1 above)
Comment (JakeWharton) [inline okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java]: If they got this far they'll see the header logged above.

Comment (swankjesse) [inline okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java]: Yeah, good point!

**Pull request merged**